V5/SpinUp/src/main.cpp

```diff
@@ -21,10 +21,101 @@ void pre_auton( void ) {
 }
 
 void autonomous( void ) {
-
+  
 }
 
 void usercontrol( void ) {
+  double driveMode = 0;
+  double driveSpeed = 0;
+  double flspeed = 0;
+  double frspeed = 0;
+  double blspeed = 0;
+  double brspeed = 0;
+  bool flfwd = true;
+  bool frfwd = true;
+  bool blfwd = true;
+  bool brfwd = true;
+  double arc = 0;
+  double angle = 0;
+  // 0 = stationary
+  // 1 = forwards
+  // 2 = backwards
+  // 3 = left
+  // 4 = right
+  // 5 = strafe
+
+  
+  while (1) {
+    double x = rc.Axis4.position(vex::percentUnits::pct);
+    double y = rc.Axis3.position(vex::percentUnits::pct);
+    arc = asin(abs(y)/sqrt(x*x + y*y));
+    if (x>=0) {
+      if (y>=0) arc = pi/2 - arc;
+      else arc = 3pi/2 + arc;
+    }
+    else {
+      if (y>=0) arc = pi/2 + arc;
+      else arc = 3pi/2 - arc;
+    }
+    angle = arc2deg(arc);
+    
+    
+
+
+    if (rc.Axis3.position(vex::percentUnits::pct) > 5) {
```

Comment:
@coco131313  Please reorganize ("refactor") your code so that within each while loop, the code reads Axis.position() only once (for each axis)